kroxylicious/src/test/java/io/kroxylicious/proxy/internal/FilterHandlerTest.java

```diff
@@ -49,6 +54,47 @@ public void testForwardRequest() {
         assertEquals(frame, propagated, "Expect it to be the frame that was sent");
     }
 
+    @Test
+    public void testFilterNotifiedWhenChannelClosed() {
```

Comment:
I wonder if mocking the filter and using verify to confirm the notification was propagated might be easier to read. 

Or maybe extracting the filter to a private class within the test case so the flow is easier to read.

Similar thoughts apply to `testFilterNotifiedWhenRemovedFromChannel` but it seems pointless to make the same comment twice.